db/migrate/20251003110349_add_withdrawn_deferred_to_training_periods.rb

```diff
@@ -0,0 +1,26 @@
+class AddWithdrawnDeferredToTrainingPeriods < ActiveRecord::Migration[8.0]
+  def change
+    create_enum :api_withdrawal_reasons, %w[
+      left-teaching-profession
+      moved-school
+      mentor-no-longer-being-mentor
+      switched-to-school-led
+      other
+    ]
+
+    create_enum :api_deferral_reasons, %w[
+      bereavement
+      long-term-sickness
+      parental-leave
+      career-break
+      other
+    ]
+
+    change_table :training_periods, bulk: true do |t|
+      t.date :api_deferred_at
+      t.enum :api_deferral_reason, enum_type: :api_deferral_reasons
+      t.date :api_withdrawn_at
```

Comment:
Should we make the type `datetime`? especially as the columns end in `_at` which usually means datetime.

Reply:
yeah looking at the API spec this should be a datetime, I'll update 👍